lib/OpenQA/WebAPI/Auth/OpenID.pm

```diff
@@ -105,9 +105,16 @@ sub auth_response {
 
     my $msg = "The openID server doesn't respond";
 
+    my $err_handler = sub {
+        my ($err, $txt) = @_;
+        $self->app->log->error($err, $txt);
+        $self->flash(error => "$err: $txt");
+        return (error => 0);
+    };
+
     $csr->handle_server_response(
         not_openid => sub {
-            die "Not an OpenID message";
+            return $err_handler->("Failed to login", "OpenID provider returned invalid data. Please retry again later");
```

Comment:
As you underlined the problem seems to lay in the OpenID provider, but Net::OpenID::Consumer will bail out if we don't cover the error https://metacpan.org/source/WROG/Net-OpenID-Consumer-1.18/lib/Net/OpenID/Consumer.pm#L440 . Looks like we hit `not_openid` when providers returns no message, or can't be parsed by the consumer. @okurz what kinda of message do you suggest here?